index.ts

```diff
@@ -143,8 +145,10 @@ export function gretch<T = DefaultGretchResponse, A = DefaultGretchError>(
         }
 
         if (response.ok) {
+          // @ts-expect-error resolved is used before being assigned
           data = resolved as T;
         } else {
+          // @ts-expect-error resolved is used before being assigned
           error = (resolved || new HTTPError(response)) as A;
```

Comment:
Older versions of TS were more lenient, it appears. Need to be careful with overrides like this, but we're being careful to check things before casting these types.